### ISSUE TITLE
Angles not working

### ISSUE THREAD

**Issue by DylanBrass** [EvHaus/react-pdf-charts#250]
### Describe the bug

Whenever you put an angle like so :
```
<XAxis dataKey="serviceName" name={"Service"}
                               interval={0} angle={-45}
                               textAnchor="end" height={200}/>
```

It will either disapear (when -90 or 90 +) or  got to the middle.

### Reproduction

- Simply add an angle on an axis or even label

### Validations

- [X] Check that there isn't [already an issue](https://github.com/EvHaus/react-pdf-charts/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that you're confident that this is a legitimate bug. To ask a question open a [GitHub Discussion](https://github.com/EvHaus/react-pdf-charts/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by EvHaus** [comment]
Confirmed. I took a look, and I'm not entirely sure what's wrong here. It almost seems like the transform origin is set incorrectly for `<Text>` elements inside `react-pdf`. You can see the labels rotate, but not relative to their element.

![image](https://github.com/EvHaus/react-pdf-charts/assets/226640/540c350e-08fc-4e6b-b139-83b4327365bd)

**Comment by DylanBrass** [comment]
Thank you, I thought I was going crazy haha, I wish I could help but I have near 0 experience with SVG's.

**Comment by urfx** [comment]
looks like you are adding an angle to a group of text items (as you said the axis) rather to each text item (ie the label)... 

btw this helped me find my issue which I can fix. (not sure I can fix yours but will revert if I do)

**Comment by kylekz** [comment]
This is going to sound really dumb, but I fixed my radar chart issue with a small pnpm patch that resets the text transform rotate to 0 for elements with a specific class.

You can see the `<PolarRadiusAxis>` ticks being completely off here:
![image](https://github.com/user-attachments/assets/6d79b65b-02d8-4053-a994-b773a7b64498)

When I set the angle to 90, it's fixed, but I don't want the angle to be in the center:
![image](https://github.com/user-attachments/assets/6de9e154-7e01-4d24-9a9e-9b7b92e812ad)

https://github.com/EvHaus/react-pdf-charts/blob/dev/src/index.tsx#L305
```ts
case "text":
  if (baseProps.transform && attribs.class ===
    "recharts-text recharts-polar-radius-axis-tick-value") {
     baseProps.transform = baseProps.transform.replace(/rotate\(60(?=,)/g, 'rotate(0')
  }
  return renderTextElement({
    baseProps,
    chartStyle,
    children: children2,
    node
});
```

![{BB9DD157-F86E-40FC-AE5B-BA272714274B}](https://github.com/user-attachments/assets/dd5230d7-af27-4150-9838-d6b4a90721ef)

This is by no means perfect, but it's close enough for me. Not sure if this helps in diagnosing the root issue.

**Comment by JElgar** [comment]
I noticed the same behavior when using a rotation on the YAxis label. I.e. 
```
<YAxis
	label={{
		value: `My y axis label`,
		style: { textAnchor: 'middle' },
		angle: -90,
		position: 'left',
		offset: 0,
	}}
/>
```
Currently the rotated label ends up on the chart rather than to the left of it.

**Comment by jontesen** [comment]
Hi,
I’ve noticed the same issue with the YAxis label. Adding angle: -90 not only rotates the text but also translates it above the Y-axis, which shouldn’t happen.

I’m wondering if there’s been any update or potential fix for this issue?

**Comment by EvHaus** [comment]
This is mostly a note to myself. Here's a minimal repro of the issue:

An `<svg>` element with some angled text in regular HTML renders them on the bottom and rotated: [REPL Link](https://codebeautify.org/htmlviewer/y25d219a5)

![Screenshot 2025-01-13 at 8 29 19 PM](https://github.com/user-attachments/assets/ff820680-5b18-40b6-83b6-0473cfc25f1a)

In `react-pdf` the same equivalent `<Svg />` code generates wrong positioning: [REPL Link](https://react-pdf.org/repl?code=3187b0760ce02e00408a057025803c450298c0bc300500943807cf805030c00f0022230080b6198531165d400a02180e659a004f003619b006f61622003a0046200099080beed3a72a019401b9f18d3c54a8a231c88fb54c00ee489540016d80110056000c1e5cc471891f23942b8033178b86a696800a860a2c080013920b140f1412382ba2941408230f9d83b38b800b184f9f80506877a19422480035b88b9838060f834f24001992632ba35a661e002d295b800d0c001b34cc0013003b08410f8a2bbccf90ab8ac8575c540020983023926b8b128faf520888ab80312cebc447144c4400038f4c26cb9b630152b83cb265801183005622f004306395000f4d11f8f5229f646c5e218a8951b10964aa5d29930365d07902ad9ec4e57394ea5540b045c157aa345aae769813a306e9f40643743a430e3498cc212159ac95e32d96ca009c337dbadfeae0954ae59ad7bca767b55a1de2a773a5c5cd75bbdd1e2e179bd7198d46fcc0aa973aba55ab94eb81bb1718321d0889c2b000212c5a2c0b6ad0a28ee1ca3e28e30248a55824ac8b87294c28d24af4cabf899b51f341d9ad2e4f2f9107ea25062e61b0b450b45878001cb210ab6dbedb6e9495ab154021bcd8ee0f5ba55d4b9f6069399c2e892b9806e303b83d9e32f7a7d34f8d0d3bfb2da1c76479ed0782a13080cc000c22187747a8919c47cf104f8d12931914da7f219e29d3598c9a8b2ea85b34c58745d2243d39602b5642a8c133d68d878dd9ac1b1ecf068ee3af24711ad3acef3a2e96b5ab32ae6b8c6f69fc7daa107b7a479fab0bf0580d097ba2f7baeb79b0cc4c471826c4abe64aa61487ed497e650feb99feaca011c9b4206f2607f295a0a2308ab04cc211b810aba6eabccb221bda846a4699a72c687ea1861a5389a6682e16b2e36bb1eba918e802aa7a99a4ca466513eb1efe9d130000a28c586764de04b5eb1bc44fa2623292e4ae402514b4b090ca89ccb890d1019cb4965856558d63058a302948b0843baee6db694b0f6c859445495a5484ee57ae86606671a33a9a73b9a4b95a2bb5e2466e00a15c55d5654f820951be89e3e4006201685ac6858f9712fb457c6c554bc559889d50a5005a592496a0781d9429b5b29057caf5a554ea94177190729993ab538675f84f5415f50eb5db7479d464df0800e2b350558946ec7227a1f0b8b2201a44c89d00c330ac3b004000dc643a3a0240b0118100e030368a6188da1f818140b230089060b58481c2282a0005c300d39a2fc4a128481807c3442037c8cea9533deacfb39ce06fc6308cfcc02cb33c1b31cdf000048260017b80690887cf4c1c2a852e18fa233cce708a2244a060892330039042df0a086080223d8302534a39b5ad90aa2a3e8c004a54f00501703414db225373a9b78150882a0e816048b101ed000000)

![Screenshot 2025-01-13 at 8 29 27 PM](https://github.com/user-attachments/assets/850fed95-56a8-4ede-a3a2-1deefc780301)

Some theories that need testing:

- It's either the `rotate()` function in the `transform` attribute that's broken in `react-pdf`, or
- It's the positioning in the `tspan` elements that isn't working. `react-pdf` is known to have limited support for `tspan` elements, or
- It would be something else too.

Needs further investigation.

**Comment by SM1512J** [comment]
Any update on this issue - I am still experiencing it?

**Comment by SM1512J** [comment]
@EvHaus  Any update on this issue - I am still experiencing it?

**Comment by EvHaus** [comment]
@SM1512J This is a bug in `react-pdf`'s engine and it's not very likely that it will be something I could fix in this library. I'm keeping the ticket open mostly so that I can try to help the `react-pdf` team isolate where the issue is at some point.

The team there did some work recently on improving `<Tspan>` support, but I just checked and looks like it didn't help resolve this problem. So more work will need to be done to handle transforms.

**Comment by SM1512J** [comment]
@EvHaus Understood, thank you. Is there currently any known alternative/work-around which could be used to create the same 90 degree angled Y-Axis Label?

**Comment by EvHaus** [comment]
Unfortunately I don't have any good suggestions here, other than to manually render the text outside the charts and position it absolutely over-top the chart, similar to how [this example](https://github.com/EvHaus/react-pdf-charts/blob/dev/examples/recharts-images.tsx#L60-L65) super-positions and image. It's a really terrible workaround and might not be viable for you, but it's the best suggestion I have at this time if you absolutely must have angled labels.

**Comment by EvHaus** [comment]
@benkraus No. Pie charts should work just fine. Here's [an example](https://stackblitz.com/edit/react-pdf-charts-starter-lvmcxufj?file=src%2FApp.tsx).
